# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit/mips64/MacroAssembler-mips64.cpp
# Commit: 3ecda60613c5
# Full Hash: 3ecda60613c5cb7a57cc7004919c7d12c3a29268
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-22 03:08:27
# Description:
#   Bug 1736531 - Fix. r=jseward
#   
#   See bug for analysis.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D129010
# ==============================================================================

diff -r 1044b04cf89b -r 3ecda60613c5 js/src/jit/mips64/MacroAssembler-mips64.cpp
--- a/js/src/jit/mips64/MacroAssembler-mips64.cpp	Thu Oct 21 15:44:59 2021 +0000
+++ b/js/src/jit/mips64/MacroAssembler-mips64.cpp	Thu Oct 21 15:46:28 2021 +0000
@@ -2288,6 +2288,11 @@
   MOZ_CRASH("IMPLEMENTME");
 }
 
+void MacroAssembler::widenInt32(Register r) {
+  // I *think* this is correct.  It may be redundant.
+  move32To64SignExtend(r, Register64(r));
+}
+
 void MacroAssembler::wasmTruncateDoubleToUInt32(FloatRegister input,
                                                 Register output,
                                                 bool isSaturating,